### PR TITLE
Add spatial plot function for 10X Visium data

### PR DESCRIPTION
**Pull request by yihming**
Function `spatial` defined in `plotting.plot_library`.

Besides, the following modifications are applied accordingly:
* Add `_plot_spots` function in `plotting.plot_utils` to make scatter plot with point circle size being data-specific.
* Add `marker_size` and `scale_factor` arguments to function `scatter` function in `plotting.plot_library`:
  * `spatial` function sets `marker_size` and `scale_factor` based on the spatial image to use as the background.
  * Use `_plot_spots` instead of `matplotlib.pyplot.scatter` for spatial plots.
  * For general scatter plots, both arguments are `None`.
*

### CODE REVIEW

**Review by bli25 — CHANGES_REQUESTED** [review]
- Line 115, Scale factor is only 10x Visium data related
- Line 215, use scale factor to judge
- Line 217-218;line 231-232, no need to test if scale_factor is not None
- Line 267, use scale_factor instead
- Line 273, use scale_factor instead
- Line 284-285, no need to test scale_factor not None
- Line 297, use scale_factor
- Line 588: expose vmin and vmax
- Line 611: may be copy doc from scatter alpha.
- Line 611: add alpha_img.
- Line 652: no passing **kwargs

**Comment by yihming** [comment]
@bli25 Changes have been made.

In `spatial` function definition, I commented that `alpha` is used for blending the **attribute layers**, while `alpha_img` for the **background spatial image**. Are these the appropriate wording?

File: pegasus/plotting/plot_library.py [hunk 1]
```diff
@@ -531,6 +580,91 @@ def scatter_groups(
...
+    attr: ``str``, optional, default: ``None``
+        Color scatter plots by attribute 'attr'. This attribute should be one key in data.obs, data.var_names (e.g. one gene) or data.obsm (attribute has the format of 'obsm_key@component', like 'X_pca@0'). If it is categorical, a palette will be used to color each category separately. Otherwise, a color map will be used.
+        If ``None``, just plot data points of the same color.
+    basis: ``str``, optional, default: ``spatial``
+        Basis to be used to generate spatial plots. Must be the 2D array showing the spatial coordinates of data points.
+    resolution: ``str``, optional, default: ``hires``
+        Use the spatial image whose value is specified in ``data.img['image_id']`` to show in background.
+        For 10X Visium data, user can either specify ``hires`` or ``lowres`` to use High or Low resolution spatial images, respectively.
+    cmaps: ``str`` or ``List[str]``, optional, default: ``viridis``
+        The colormap(s) for plotting numeric attributes. The default ``viridis`` colormap theme follows the spatial plot function in SCANPY (``scanpy.pl.spatial``).
```
Comment (bli25) [inline pegasus/plotting/plot_library.py:613]: Please add doc for vmin and vmax here.

File: pegasus/plotting/plot_library.py [hunk 2]
```diff
@@ -256,7 +305,7 @@ def scatter(
                                 ncol=_get_legend_ncol(label_size, legend_ncol),
                             )
                             for handle in legend.legendHandles:
-                                handle.set_sizes([300.0])
+                                handle.set_sizes([300.0 if marker_size is None else 100.0])
```
Comment (bli25) [inline pegasus/plotting/plot_library.py]: Please use scale_factor

**Review by bli25 — CHANGES_REQUESTED** [review]

**Review by bli25 — APPROVED** [review]
